docs/runtime/clr-metrics.md

```diff
@@ -0,0 +1,324 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: CLR
+--->
+
+# Semantic Conventions for .NET CLR Metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+This document describes semantic conventions for .NET CLR runtime metrics in OpenTelemetry.
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [CLR Garbage Collection](#clr-garbage-collection)
+  - [Metric: `clr.gc.collections.count`](#metric-clrgccollectionscount)
+  - [Metric: `clr.gc.objects.size`](#metric-clrgcobjectssize)
+  - [Metric: `clr.gc.allocations.size`](#metric-clrgcallocationssize)
+  - [Metric: `clr.gc.committed_memory.size`](#metric-clrgccommitted_memorysize)
+  - [Metric: `clr.gc.heap.size`](#metric-clrgcheapsize)
+  - [Metric: `clr.gc.heap.fragmentation.size`](#metric-clrgcheapfragmentationsize)
+  - [Metric: `clr.gc.duration`](#metric-clrgcduration)
+- [Just-In-Time (JIT) Compiler](#just-in-time-jit-compiler)
+  - [Metric: `clr.jit.il_compiled.size`](#metric-clrjitil_compiledsize)
+  - [Metric: `clr.jit.methods_compiled.count`](#metric-clrjitmethods_compiledcount)
+  - [Metric: `clr.jit.compilation_time`](#metric-clrjitcompilation_time)
+- [Thread pool](#thread-pool)
+  - [Metric: `clr.thread_pool.threads.count`](#metric-clrthread_poolthreadscount)
+  - [Metric: `clr.thread_pool.work_items.count`](#metric-clrthread_poolwork_itemscount)
+  - [Metric: `clr.thread_pool.queue.length`](#metric-clrthread_poolqueuelength)
+- [General](#general)
+  - [Metric: `clr.monitor.lock_contention.count`](#metric-clrmonitorlock_contentioncount)
+  - [Metric: `clr.timer.count`](#metric-clrtimercount)
+  - [Metric: `clr.assemblies.count`](#metric-clrassembliescount)
+  - [Metric: `clr.exceptions.count`](#metric-clrexceptionscount)
+
+<!-- tocstop -->
+
+## CLR Garbage Collection
+
+**Status**: [Experimental][DocumentStatus]
+
+**Description:** .NET Common Language Runtime (CLR) metrics relating to garbage collection, captured under the namespace `clr.gc.*`.
+
+### Metric: `clr.gc.collections.count`
+
+This metric is [recommended][MetricRecommended].
+This metric is obtained from [`GC.CollectionCount(int generation)`](https://learn.microsoft.com/en-us/dotnet/api/system.gc.collectioncount).
+
+<!-- semconv metric.clr.gc.collections.count(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `clr.gc.collections.count` | Counter | `{collection}` | Number of garbage collections that have occurred since the process started.
```

Comment:
Why not use observable instruments in such cases? Do you expect the runtime will publish this using timer?

Reply:
@tarekgh The semantic conventions [prefer using synchronous instrument names](https://opentelemetry.io/docs/specs/semconv/general/metrics/#instrument-types) even though the implementation may then choose the async (observable) variant.